[CI] RepositoryAnalysisFailureIT testFailsIfEmptyRegisterRejected failing

**Build Scans:**
- [elasticsearch-periodic #3558 / openjdk22_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/epujyfycs6h5a)
- [elasticsearch-periodic-platform-support #3464 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/adft43icrc7qe)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:snapshot-repo-test-kit:internalClusterTest" --tests "org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisFailureIT.testFailsIfEmptyRegisterRejected" -Dtests.seed=BC8CB7E68FD6CB74 -Dtests.locale=hu-HU -Dtests.timezone=AET -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testFailsIfEmptyRegisterRejected'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisFailureIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a string containing "Elasticsearch observed the storage system underneath this repository behaved incorrectly" and a string containing "not suitable for use with Elasticsearch snapshots" and a string containing "https://www.elastic.co/guide/en/elasticsearch/reference/master/repo-analysis-api.html")
     but: a string containing "Elasticsearch observed the storage system underneath this repository behaved incorrectly" was "[test-repo] Repository analysis timed out. Consider specifying a longer timeout using the [?timeout] request parameter. See [https://www.elastic.co/guide/en/elasticsearch/reference/master/repo-analysis-api.html] for more information. Elasticsearch attempted to remove the data it wrote at [temp-analysis-isBLdGzaQzyPhmuf64cIkw] but may have left some behind. If so, please now remove this data manually."
```

**Issue Reasons:**
- [main] 2 failures in test testFailsIfEmptyRegisterRejected (0.3% fail rate in 655 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java
@@ -22,6 +22,7 @@ import org.elasticsearch.common.blobstore.support.BlobMetadata;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.io.stream.BytesStreamOutput;
 import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.util.BigArrays;
 import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
@@ -40,7 +41,6 @@ import org.elasticsearch.repositories.RepositoryMissingException;
 import org.elasticsearch.repositories.RepositoryVerificationException;
 import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
 import org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase;
-import org.elasticsearch.test.junit.annotations.TestIssueLogging;
 import org.elasticsearch.xcontent.NamedXContentRegistry;
 import org.elasticsearch.xpack.core.LocalStateCompositeXPackPlugin;
 import org.junit.Before;
@@ -58,12 +58,14 @@ import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
+import static org.elasticsearch.indices.recovery.RecoverySettings.INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING;
+import static org.elasticsearch.repositories.blobstore.BlobStoreRepository.MAX_RESTORE_BYTES_PER_SEC;
+import static org.elasticsearch.repositories.blobstore.BlobStoreRepository.MAX_SNAPSHOT_BYTES_PER_SEC;
 import static org.elasticsearch.repositories.blobstore.testkit.ContendedRegisterAnalyzeAction.bytesFromLong;
 import static org.elasticsearch.repositories.blobstore.testkit.ContendedRegisterAnalyzeAction.longFromBytes;
 import static org.hamcrest.Matchers.allOf;
@@ -87,9 +89,26 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         return List.of(TestPlugin.class, LocalStateCompositeXPackPlugin.class, SnapshotRepositoryTestKit.class);
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal, otherSettings))
+            // no throttling, so that even analyses which run to completion do not take too long
+            .put(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.getKey(), ByteSizeValue.ZERO)
+            .build();
+    }
+
     @Before
     public void createBlobStore() {
-        createRepositoryNoVerify("test-repo", TestPlugin.DISRUPTABLE_REPO_TYPE);
+        createRepository(
+            "test-repo",
+            TestPlugin.DISRUPTABLE_REPO_TYPE,
+            randomRepositorySettings()
+                // no throttling, so that even analyses which run to completion do not take too long
+                .put(MAX_SNAPSHOT_BYTES_PER_SEC.getKey(), ByteSizeValue.ZERO)
+                .put(MAX_RESTORE_BYTES_PER_SEC.getKey(), ByteSizeValue.ZERO),
+            false
+        );
 
         blobStore = new DisruptableBlobStore();
         for (final RepositoriesService repositoriesService : internalCluster().getInstances(RepositoriesService.class)) {
@@ -105,8 +124,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         final RepositoryAnalyzeAction.Request request = new RepositoryAnalyzeAction.Request("test-repo");
         request.blobCount(1);
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
-
-        analyseRepository(request);
+        safeAwait((ActionListener<RepositoryAnalyzeAction.Response> l) -> analyseRepository(request, l));
     }
 
     public void testFailsOnReadError() {
@@ -125,7 +143,8 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        final Exception exception = expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        final Exception exception = analyseRepository(request);
+        assertAnalysisFailureMessage(exception.getMessage());
         final IOException ioException = (IOException) ExceptionsHelper.unwrap(exception, IOException.class);
         assert ioException != null : exception;
         assertThat(ioException.getMessage(), equalTo("simulated"));
@@ -149,7 +168,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsOnChecksumMismatch() {
@@ -180,7 +199,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsOnWriteException() {
@@ -201,7 +220,8 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
 
         });
 
-        final Exception exception = expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        final Exception exception = analyseRepository(request);
+        assertAnalysisFailureMessage(exception.getMessage());
         final IOException ioException = (IOException) ExceptionsHelper.unwrap(exception, IOException.class);
         assert ioException != null : exception;
         assertThat(ioException.getMessage(), equalTo("simulated"));
@@ -223,7 +243,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
 
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsOnListingException() {
@@ -243,7 +263,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsOnDeleteException() {
@@ -258,7 +278,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsOnIncompleteDelete() {
@@ -286,7 +306,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsIfBlobCreatedOnAbort() {
@@ -303,12 +323,21 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        try {
-            analyseRepository(request);
-            assertFalse(writeWasAborted.get());
-        } catch (RepositoryVerificationException e) {
-            assertTrue(writeWasAborted.get());
-        }
+        safeAwait((ActionListener<RepositoryAnalyzeAction.Response> l) -> analyseRepository(request, l.delegateResponse((ll, e) -> {
+            if (ExceptionsHelper.unwrapCause(e) instanceof RepositoryVerificationException repositoryVerificationException) {
+                assertAnalysisFailureMessage(repositoryVerificationException.getMessage());
+                assertTrue(
+                    "did not abort write, so why did the verification fail?",
+                    // clear flag for final assertion
+                    writeWasAborted.compareAndSet(true, false)
+                );
+                ll.onResponse(null);
+            } else {
+                ll.onFailure(e);
+            }
+        })));
+
+        assertFalse(writeWasAborted.get());
     }
 
     public void testFailsIfRegisterIncorrect() {
@@ -325,7 +354,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return register.compareAndExchange(expected, updated);
             }
         });
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
     }
 
     public void testFailsIfRegisterHoldsSpuriousValue() {
@@ -355,14 +384,22 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return register.compareAndExchange(expected, updated);
             }
         });
-        try {
-            analyseRepository(request);
-            assertFalse(sawSpuriousValue.get());
-        } catch (RepositoryVerificationException e) {
-            if (sawSpuriousValue.get() == false) {
-                fail(e, "did not see spurious value, so why did the verification fail?");
+
+        safeAwait((ActionListener<RepositoryAnalyzeAction.Response> l) -> analyseRepository(request, l.delegateResponse((ll, e) -> {
+            if (ExceptionsHelper.unwrapCause(e) instanceof RepositoryVerificationException repositoryVerificationException) {
+                assertAnalysisFailureMessage(repositoryVerificationException.getMessage());
+                assertTrue(
+                    "did not see spurious value, so why did the verification fail?",
+                    // clear flag for final assertion
+                    sawSpuriousValue.compareAndSet(true, false)
+                );
+                ll.onResponse(null);
+            } else {
+                ll.onFailure(e);
             }
-        }
+        })));
+
+        assertFalse(sawSpuriousValue.get());
     }
 
     private static void assertAnalysisFailureMessage(String message) {
@@ -387,7 +424,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return isContendedRegisterKey(key) == false;
             }
         });
-        final var exception = expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        final var exception = analyseRepository(request);
         assertThat(
             exception.getMessage(),
             allOf(
@@ -409,7 +446,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return false;
             }
         });
-        final var exception = expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        final var exception = analyseRepository(request);
         assertAnalysisFailureMessage(exception.getMessage());
         assertThat(
             asInstanceOf(RepositoryVerificationException.class, ExceptionsHelper.unwrapCause(exception.getCause())).getMessage(),
@@ -417,10 +454,6 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         );
     }
 
-    @TestIssueLogging(
-        issueUrl = "https://github.com/elastic/elasticsearch/issues/111343",
-        value = "org.elasticsearch.repositories.blobstore.testkit:TRACE"
-    )
     public void testFailsIfEmptyRegisterRejected() {
         final RepositoryAnalyzeAction.Request request = new RepositoryAnalyzeAction.Request("test-repo");
         blobStore.setDisruption(new Disruption() {
@@ -429,7 +462,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return false;
             }
         });
-        final var exception = expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        final var exception = analyseRepository(request);
         assertAnalysisFailureMessage(exception.getMessage());
         final var cause = ExceptionsHelper.unwrapCause(exception.getCause());
         if (cause instanceof IOException ioException) {
@@ -445,8 +478,18 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         }
     }
 
-    private void analyseRepository(RepositoryAnalyzeAction.Request request) {
-        client().execute(RepositoryAnalyzeAction.INSTANCE, request).actionGet(5L, TimeUnit.MINUTES);
+    private RepositoryVerificationException analyseRepository(RepositoryAnalyzeAction.Request request) {
+        return asInstanceOf(
+            RepositoryVerificationException.class,
+            ExceptionsHelper.unwrapCause(safeAwaitFailure(RepositoryAnalyzeAction.Response.class, l -> analyseRepository(request, l)))
+        );
+    }
+
+    private void analyseRepository(RepositoryAnalyzeAction.Request request, ActionListener<RepositoryAnalyzeAction.Response> listener) {
+        client().execute(RepositoryAnalyzeAction.INSTANCE, request, listener.delegateFailureAndWrap((l, response) -> {
+            RepositoryAnalysisSuccessIT.assertNoThrottling(response);
+            l.onResponse(response);
+        }));
     }
 
     private static void assertPurpose(OperationPurpose purpose) {

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java
@@ -143,7 +143,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        final Exception exception = analyseRepository(request);
+        final Exception exception = analyseRepositoryExpectFailure(request);
         assertAnalysisFailureMessage(exception.getMessage());
         final IOException ioException = (IOException) ExceptionsHelper.unwrap(exception, IOException.class);
         assert ioException != null : exception;
@@ -168,7 +168,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsOnChecksumMismatch() {
@@ -199,7 +199,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsOnWriteException() {
@@ -220,7 +220,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
 
         });
 
-        final Exception exception = analyseRepository(request);
+        final Exception exception = analyseRepositoryExpectFailure(request);
         assertAnalysisFailureMessage(exception.getMessage());
         final IOException ioException = (IOException) ExceptionsHelper.unwrap(exception, IOException.class);
         assert ioException != null : exception;
@@ -243,7 +243,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
 
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsOnListingException() {
@@ -263,7 +263,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsOnDeleteException() {
@@ -278,7 +278,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsOnIncompleteDelete() {
@@ -306,7 +306,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
             }
         });
 
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsIfBlobCreatedOnAbort() {
@@ -354,7 +354,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return register.compareAndExchange(expected, updated);
             }
         });
-        assertAnalysisFailureMessage(analyseRepository(request).getMessage());
+        assertAnalysisFailureMessage(analyseRepositoryExpectFailure(request).getMessage());
     }
 
     public void testFailsIfRegisterHoldsSpuriousValue() {
@@ -424,7 +424,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return isContendedRegisterKey(key) == false;
             }
         });
-        final var exception = analyseRepository(request);
+        final var exception = analyseRepositoryExpectFailure(request);
         assertThat(
             exception.getMessage(),
             allOf(
@@ -446,7 +446,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return false;
             }
         });
-        final var exception = analyseRepository(request);
+        final var exception = analyseRepositoryExpectFailure(request);
         assertAnalysisFailureMessage(exception.getMessage());
         assertThat(
             asInstanceOf(RepositoryVerificationException.class, ExceptionsHelper.unwrapCause(exception.getCause())).getMessage(),
@@ -462,7 +462,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
                 return false;
             }
         });
-        final var exception = analyseRepository(request);
+        final var exception = analyseRepositoryExpectFailure(request);
         assertAnalysisFailureMessage(exception.getMessage());
         final var cause = ExceptionsHelper.unwrapCause(exception.getCause());
         if (cause instanceof IOException ioException) {
@@ -478,7 +478,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         }
     }
 
-    private RepositoryVerificationException analyseRepository(RepositoryAnalyzeAction.Request request) {
+    private RepositoryVerificationException analyseRepositoryExpectFailure(RepositoryAnalyzeAction.Request request) {
         return asInstanceOf(
             RepositoryVerificationException.class,
             ExceptionsHelper.unwrapCause(safeAwaitFailure(RepositoryAnalyzeAction.Response.class, l -> analyseRepository(request, l)))

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
@@ -10,6 +10,7 @@ package org.elasticsearch.repositories.blobstore.testkit;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.cluster.metadata.RepositoryMetadata;
 import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.blobstore.BlobContainer;
 import org.elasticsearch.common.blobstore.BlobPath;
 import org.elasticsearch.common.blobstore.BlobStore;
@@ -26,7 +27,6 @@ import org.elasticsearch.common.util.BigArrays;
 import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
 import org.elasticsearch.core.CheckedConsumer;
 import org.elasticsearch.core.Nullable;
-import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.indices.recovery.RecoverySettings;
 import org.elasticsearch.plugins.Plugin;
@@ -38,6 +38,7 @@ import org.elasticsearch.repositories.RepositoryMissingException;
 import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
 import org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase;
 import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.ObjectPath;
 import org.elasticsearch.xpack.core.LocalStateCompositeXPackPlugin;
 import org.junit.Before;
 
@@ -51,15 +52,18 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.Semaphore;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
+import static org.elasticsearch.indices.recovery.RecoverySettings.INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING;
+import static org.elasticsearch.repositories.blobstore.BlobStoreRepository.MAX_RESTORE_BYTES_PER_SEC;
+import static org.elasticsearch.repositories.blobstore.BlobStoreRepository.MAX_SNAPSHOT_BYTES_PER_SEC;
 import static org.elasticsearch.repositories.blobstore.testkit.ContendedRegisterAnalyzeAction.longFromBytes;
 import static org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisFailureIT.isContendedRegisterKey;
+import static org.elasticsearch.test.XContentTestUtils.convertToMap;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.equalTo;
@@ -83,9 +87,21 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {
         return List.of(TestPlugin.class, LocalStateCompositeXPackPlugin.class, SnapshotRepositoryTestKit.class);
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal, otherSettings))
+            .put(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.getKey(), ByteSizeValue.ZERO)
+            .build();
+    }
+
     public void testRepositoryAnalysis() {
 
-        final Settings.Builder settings = Settings.builder();
+        final Settings.Builder settings = Settings.builder()
+            // no throttling
+            .put(MAX_SNAPSHOT_BYTES_PER_SEC.getKey(), ByteSizeValue.ZERO)
+            .put(MAX_RESTORE_BYTES_PER_SEC.getKey(), ByteSizeValue.ZERO);
+
         if (randomBoolean()) {
             settings.put(BASE_PATH_SETTING_KEY, randomAlphaOfLength(10));
         }
@@ -136,11 +152,22 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {
             blobStore.setMaxTotalBlobSize(request.getMaxTotalDataSize().getBytes());
         }
 
-        request.timeout(TimeValue.timeValueSeconds(20));
-
-        client().execute(RepositoryAnalyzeAction.INSTANCE, request).actionGet(30L, TimeUnit.SECONDS);
+        request.timeout(SAFE_AWAIT_TIMEOUT);
+        final RepositoryAnalyzeAction.Response response = safeAwait(l -> client().execute(RepositoryAnalyzeAction.INSTANCE, request, l));
 
         assertThat(blobStore.currentPath, nullValue());
+
+        assertNoThrottling(response);
+    }
+
+    static void assertNoThrottling(RepositoryAnalyzeAction.Response response) {
+        try {
+            final var responseMap = convertToMap(response);
+            assertEquals(Strings.toString(response), 0, (int) ObjectPath.eval("summary.write.total_throttled_nanos", responseMap));
+            assertEquals(Strings.toString(response), 0, (int) ObjectPath.eval("summary.read.total_throttled_nanos", responseMap));
+        } catch (IOException e) {
+            fail(e);
+        }
     }
 
     public static class TestPlugin extends Plugin implements RepositoryPlugin {

